### PR TITLE
Prepare CI/CD Workflows

### PR DESCRIPTION
**Pull request by mbfm**
Hello,
we're preparing our CI/CD contributions and want to use a few variables to control that.

Additionally, we've changed the `development` branch to not be deleteable. We want to use it long-term as a basin for stabilizing before merging onto `main`.

And I've changed the spelling for the Org name. We seem to have standardized on that specific casing.

### CODE REVIEW

File: otterdog/eclipse-opendut.jsonnet
Comment (netomi) [inline otterdog/eclipse-opendut.jsonnet]: this should be variables+:

File: otterdog/eclipse-opendut.jsonnet
Comment (netomi) [inline otterdog/eclipse-opendut.jsonnet]: I assume you want a branch protection rule that only prevents deletion of the branch?
This rule will not do exactly what you want as it will use the default values, you can go to the playground to adjust the rule to your needs: https://eclipse-opendut.github.io/.eclipsefdn/playground/

Here is an example:

image

you see that various setting are there even if you dont specify them, but you can change them ofc by overriding them in your snippet.

**Comment by netomi** [comment]
I would suggest that you update your GitHub profile with a bit more information about yourself, I can see that you are a committer to the project but have no clue to which person that GitHub handle links.

File: otterdog/eclipse-opendut.jsonnet
Comment (mbfm) [reply otterdog/eclipse-opendut.jsonnet]: Thanks, I've changed it accordingly.
The documentation here appears to be wrong then: https://otterdog.readthedocs.io/en/latest/reference/organization/repository/variable/#example-usage

File: otterdog/eclipse-opendut.jsonnet
Comment (kKdH) [reply otterdog/eclipse-opendut.jsonnet]: Are you sure that the `allows_deletions` setting works as expected? Because i can delete any branch:
Screenshot_20231214_144518

Comment (mbfm) [reply otterdog/eclipse-opendut.jsonnet]: Ah, I had not realized that was the default. I've removed that commit for now.

In addition to what @kKdH already commented, we similarly face the issue that the branch can be deleted after a pull request was merged from `development` to `main`:

Screenshot_20231214_144017

After pressing the button, it looks like this and the branch is gone:

Screenshot_20231214_144038

**Comment by mbfm** [comment]
I added two more changes (regarding merge commits and status checks) after talking with @kKdH.

File: otterdog/eclipse-opendut.jsonnet
Comment (netomi) [reply otterdog/eclipse-opendut.jsonnet]: the changes will only become active when this PR got merged and applied, till now there is no branch protection rule in place to prevent deletion of the development branch.

Comment (netomi) [reply otterdog/eclipse-opendut.jsonnet]: so, yes you will need to add a branch protection rule for the development branch that prevents deletion. Your rule was ok I was just pointing out that apart from the setting the rule will also have some default values e.g. number of approvals of 2 which you might not want for the development branch so you have to adjust these settings with the playground.

Comment (kKdH) [reply otterdog/eclipse-opendut.jsonnet]: Ah. We misunderstood the meaning of 'default' in this case. :see_no_evil: Sorry. We thought that there is a branch protection rule (a default one) which prevents the deletion of a branch.

Thank you for your patience!

Comment (netomi) [reply otterdog/eclipse-opendut.jsonnet]: no worries, I am here to help and I understand it might not be obvious from the start. Once you get the knack its actually a pretty handy tool imho.

Comment (mbfm) [reply otterdog/eclipse-opendut.jsonnet]: I've re-added the branch protection rule. Please check, if this now looks plausible.

**Comment by netomi** [comment]
in general LGTM, 2 remarks:

- we usually wanna prevent allows_force_pushes
- when disabling requires_pull_request,  required_approving_review_count should be set to null instead of 0 but that's just a minor nitpick and how GitHub models dependent parameters

**Comment by mbfm** [comment]
>     * we usually wanna prevent allows_force_pushes

We'd like to start out with it for now. We'll see how it works out.

>     * when disabling requires_pull_request,  required_approving_review_count should be set to null instead of 0 but that's just a minor nitpick and how GitHub models dependent parameters

I have changed this accordingly.

**Comment by netomi** [comment]
is the PR ready to be integrated or are you still planning on making changes?

**Comment by mbfm** [comment]
The PR can be integrated, yes.

**Review by netomi — APPROVED** [review]

**Comment by netomi** [comment]
changes are live.

**Comment by mbfm** [comment]
Thanks. Unfortunately, the `eclipse-eca-validation:eclipsefdn/eca` status check on the development branch seems to be causing problems. Our pushes are getting rejected.

We figured, it would make sense to leave this check enabled, to reject changes without appropriate ECA as early as possible. Was this a wrong assumption or does the ECA check not succeed for some reason? It seemed to succeed for me when posting the pull requests here...

**Comment by netomi** [comment]
ah ok, I know whats going on. The eca check is only performed when a PR is openend and some service adds this check. When you define a required status check via a branch protection rule, it is checked *ALWAYS* so even if you just push directly to the branch. As there will be nobody to attach the eca check to the commit, you cant push the commit.

We solved that for some other project using repository rulesets, which are basically the same as branch protection rules but work more cleanly. Take a look here:

https://github.com/jetty/.eclipsefdn/blob/main/otterdog/jetty.jsonnet#L367

I would suggest if you wanna push to the development branch directly but still require the eca check for PRs (which is a good thing), then convert the branch protection rule to a repository ruleset. I can help with that ofc.

**Comment by mbfm** [comment]
That does sound like what we want. If you happen to know what's required here, we would certainly appreciate a suggestion. I can't tell immediately what makes it work, although I could take a deeper look at it tomorrow...

**Pull request merged**
